core/src/main/java/org/elasticsearch/index/mapper/core/CompletionFieldMapper.java

```diff
@@ -18,151 +18,98 @@
-import org.elasticsearch.search.suggest.completion.CompletionTokenStream;
-import org.elasticsearch.search.suggest.context.ContextBuilder;
-import org.elasticsearch.search.suggest.context.ContextMapping;
-import org.elasticsearch.search.suggest.context.ContextMapping.ContextConfig;
+import org.elasticsearch.index.mapper.*;
+import org.elasticsearch.index.mapper.object.ArrayValueMapperParser;
+import org.elasticsearch.search.suggest.completion.CompletionSuggester;
+import org.elasticsearch.search.suggest.completion.context.ContextMappings;
 
 import java.io.IOException;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Set;
-import java.util.SortedMap;
-
-import static org.elasticsearch.index.mapper.MapperBuilders.completionField;
+import java.util.*;
+
+import static org.elasticsearch.index.mapper.MapperBuilders.*;
 import static org.elasticsearch.index.mapper.core.TypeParsers.parseMultiField;
 
 /**
+ * Mapper for completion field. The field values are indexed as a weighted FST for
+ * fast auto-completion/search-as-you-type functionality.<br>
  *
+ * Type properties:<br>
+ * <ul>
+ *  <li>"index_analyzer": "simple", (default)</li>
+ *  <li>"search_analyzer": "simple", (default)</li>
+ *  <li>"preserve_separators" : true, (default)</li>
+ *  <li>"preserve_position_increments" : true (default)</li>
+ *  <li>"min_input_length": 50 (default)</li>
+ *  <li>"contexts" : CONTEXTS</li>
+ * </ul>
+ * see {@link ContextMappings#load(Object, Version)} for CONTEXTS<br>
+ * see {@link #parse(ParseContext)} for acceptable inputs for indexing<br>
+ * <p>
+ *  This field type constructs completion queries that are run
+ *  against the weighted FST index by the {@link CompletionSuggester}.
+ *  This field can also be extended to add search criteria to suggestions
+ *  for query-time filtering and boosting (see {@link ContextMappings}
  */
-public class CompletionFieldMapper extends FieldMapper {
+public class CompletionFieldMapper extends FieldMapper implements ArrayValueMapperParser {
 
     public static final String CONTENT_TYPE = "completion";
 
     public static class Defaults {
-        public static final CompletionFieldType FIELD_TYPE = new CompletionFieldType();
-
+        public static final MappedFieldType FIELD_TYPE = new CompletionFieldType();
         static {
             FIELD_TYPE.setOmitNorms(true);
             FIELD_TYPE.freeze();
         }
-
         public static final boolean DEFAULT_PRESERVE_SEPARATORS = true;
         public static final boolean DEFAULT_POSITION_INCREMENTS = true;
-        public static final boolean DEFAULT_HAS_PAYLOADS = false;
         public static final int DEFAULT_MAX_INPUT_LENGTH = 50;
     }
 
     public static class Fields {
         // Mapping field names
-        public static final String ANALYZER = "analyzer";
+        public static final ParseField INDEX_ANALYZER = new ParseField("index_analyzer", "analyzer");
```

Comment:
This should stay just `analyzer`. Behavior was changed for all field mappers to have just `analyzer`, and `search_analyzer` simply overrides the search time behavior.